packages/client/auth/src/CrossmintAuthClient.ts

```diff
@@ -0,0 +1,182 @@
+import type { UseSignInData } from "@farcaster/auth-kit";
+import {
+    type AuthMaterialWithUser,
+    CROSSMINT_API_VERSION,
+    CrossmintAuth,
+    type OAuthProvider,
+    REFRESH_TOKEN_PREFIX,
+    SESSION_PREFIX,
+} from "@crossmint/common-sdk-auth";
+import type { Crossmint } from "@crossmint/common-sdk-base";
+import { type CancellableTask, queueTask } from "@crossmint/client-sdk-base";
+import { deleteCookie, getCookie, getJWTExpiration, setCookie, TIME_BEFORE_EXPIRING_JWT_IN_SECONDS } from "./utils";
+
+export class CrossmintAuthClient extends CrossmintAuth {
+    private callbacks: CrossmintAuthClientCallbacks;
+    private refreshTask: CancellableTask | null = null;
+    private isRefreshing = false;
+
+    private constructor(crossmint: Crossmint, callbacks: CrossmintAuthClientCallbacks = {}) {
+        super(crossmint);
+        this.callbacks = callbacks;
+        this.handleRefreshToken();
+    }
+
+    public static from(crossmint: Crossmint, callbacks?: CrossmintAuthClientCallbacks): CrossmintAuthClient {
+        return new CrossmintAuthClient(crossmint, callbacks);
+    }
+
+    public getSession() {
+        this.handleRefreshToken();
+        return getCookie(SESSION_PREFIX);
```

Comment:
Without an `await` in the previous line, we are not guaranteeing that the cookie is already set here.

Reply:
This is so we can mimick the API decided for the server sdk but in most cases it won't be used like this. We only expect this to return the jwt. `handleRefreshToken` triggers a task to get the most recent auth material and set off a timer to re-check later on so it's more like a background process that we want to make sure it's being called on init. Right now we're sticking to triggering this when creating an instance in the constructor

Reply:
But what's the expected use if this returns `null` because the cookie is not set yet?
We've already triggered a `refresh` in the background, but the developer doesn't have any `Promise` to wait for it.

Reply:
I think I'll just delete it. This is a half-baked method just to be API compliant with server sdk. It returns the jwt if it's there for immediate use as we needed that behaviour in the react sdk, but I couldn't find a way to substitute calling `handleRefreshToken` on init and calling it here instead in a way that I could make it work sensibly with the current sdk. So I left the function half-baked, and maintained the other behaviour that uses callbacks. If we followed the way the server sdk does it, we wouldn't return the jwt until it has been validated and refreshed.